source/client-side-encryption/client-side-encryption.rst

```diff
@@ -740,14 +761,14 @@ ClientEncryption
 
       // Creates a new key document and inserts into the key vault collection.
       // Returns the _id of the created document as a UUID (BSON binary subtype 0x04).
-      createKey(kmsProvider: String, opts: Optional<DataKeyOpts>): Binary;
+      createKey(kmsProvider: KMSProviderName, opts: DataKeyOpts | null): Binary;
```

Comment:
`createKey` is expected to accept arbitrary strings for the `kmsProvider` for forward compatibility. If a KMS provider is added in the future, users may only need to upgrade libmongocrypt, not the driver, to use it.

Does using `KMSProviderName` as the type for `kmsProvider` suggest a driver may not allow arbitrary strings? If so, I suggest using `string` for `kmsProvider`.

Reply:
I've clarified the intention of `KMSProviderName`, which should hopefully avoid any possible confusion.